notebooks/retriever-reader-pipelines/prompt_reader_LLM.ipynb

```diff
@@ -0,0 +1,528 @@
+    "from primeqa.pipelines.qa_pipeline import QAPipeline\n",
+    "import json"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[Feb 16, 20:52:11] #> base_config.py from_path /dccstor/colbert-ir/franzm/indexes/oct2_10_11/oct2_10_11_exp/indexes//oct2_10_11_indname/metadata.json\n",
+      "[Feb 16, 20:52:11] #> base_config.py from_path args loaded! \n",
+      "[Feb 16, 20:52:11] #> base_config.py from_path args replaced ! \n",
+      "[Feb 16, 20:52:15] #>>>>> at ColBERT name (model type) : /dccstor/colbert-ir/franzm/experiments/oct2_7_12_1.5e-06/none/2022-10/09/15.21.39/checkpoints/colbert.dnn.batch_91287.model\n",
+      "[Feb 16, 20:52:15] #>>>>> at BaseColBERT name (model type) : /dccstor/colbert-ir/franzm/experiments/oct2_7_12_1.5e-06/none/2022-10/09/15.21.39/checkpoints/colbert.dnn.batch_91287.model\n",
+      "[Feb 16, 20:52:23] factory model type: xlm-roberta-large\n",
+      "[Feb 16, 20:52:39] get query model type: xlm-roberta-large\n",
+      "[Feb 16, 20:52:40] get doc model type: xlm-roberta-large\n",
+      "[Feb 16, 20:52:45] #> Loading codec...\n",
+      "[Feb 16, 20:52:45] #> base_config.py from_path /dccstor/colbert-ir/franzm/indexes/oct2_10_11/oct2_10_11_exp/indexes//oct2_10_11_indname/metadata.json\n",
+      "[Feb 16, 20:52:45] #> base_config.py from_path args loaded! \n",
+      "[Feb 16, 20:52:45] #> base_config.py from_path args replaced ! \n",
+      "[Feb 16, 20:52:45] Loading decompress_residuals_cpp extension (set COLBERT_LOAD_TORCH_EXTENSION_VERBOSE=True for more info)...\n",
+      "[Feb 16, 20:52:46] Loading packbits_cpp extension (set COLBERT_LOAD_TORCH_EXTENSION_VERBOSE=True for more info)...\n",
+      "[Feb 16, 20:52:46] #> Loading IVF...\n"
+     ]
+    }
+   ],
+   "source": [
+    "# setup ColBERT index\n",
+    "index_root = \"/dccstor/colbert-ir/franzm/indexes/oct2_10_11/oct2_10_11_exp/indexes/\"\n",
+    "index_name = \"oct2_10_11_indname\"\n",
+    "collection = \"/dccstor/avi7/neural_ir/colbert/data/psgs_w100.tsv\"\n",
+    "\n",
+    "\n",
+    "retriever = ColBERTRetriever(index_root = index_root, \n",
+    "                                     index_name = index_name, \n",
+    "                                     collection = collection, \n",
+    "                                     max_num_documents = 3)\n",
+    "retriever.load()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "checkpoint = \"/dccstor/colbert-ir/franzm/experiments/oct2_18_10/qry_encoder\"\n",
+    "index_root = \"/dccstor/colbert-ir/franzm/experiments/\"\n",
+    "index_name = \"oct2_18_20\"\n",
+    "collection = \"/dccstor/avi7/neural_ir/colbert/data/psgs_w100.tsv\"\n",
+    "\n",
+    "retriever = DPRRetriever(index_root = index_root, \n",
```

Comment:
isn't it overwriting the ColBERT retriever ? WHy do we need the ColBERT retriever then ? Can re remove the import and instantiation of colBERTRetriever then ?

Reply:
I wanted to have the option in the notebook to run with different retrieval methods and indices. We have 2 indexes for wikipedia: the one with colBERT and the one with DPR. The user should chose one or the other.  The DPR one needs less memory but the DPR retrieval is not as good as ColBERT.

Reply:
After talking to @seirasto we decided to take out the DPR index. We were not using it in the notebook anyway.